bcgov/workbc-main
# [WR-270] configuration split

Configuration Split Implementation

5 environments have been set up

- **Dev** - local docker development environment
- **AWS Dev** - AWS development environment
- **AWS Test** - AWS test environment
- **AWS Prod** - AWS production environment
- **Local** - local environment allowing individual developers to set up custom configuration.  contents of this folder are ignored by git. Use of the Local split should not be needed and has been added for possible future use should it become necessary.

The following variable has been added to the `workbc-main/.env` file
`PROJECT_ENVIRONMENT=dev`


As configured the configuration is setup for all environments and developers do not need to do anything beyond continuing to use `drush cim` and `drush cex` as before.


**Local**, **Dev** and **AWS-Dev** environments have the **Devel** module enabled.

**AWS-Test** and **AWS-Prod** environments have the **Devel** module disabled.

**AWS-Dev**, **AWS-Test** and **AWS-Prod** environments have been configured to ignore **Backup and Migrate** module configuration


How to Test
after checking out branch and running `sync.sh` script, ensure that the Dev configuration split is active.

Confirm the following:

- appropriate environment is set by going to `http://workbc.docker.localhost:8000/admin/config/development/configuration/config-split` The **Dev** listing should have a status of `active (overwritten)`
- Devel module should be enabled
- Configuration Ignore setting  (`http://workbc.docker.localhost:8000/admin/config/development/configuration/ignore`) should be empty

change to AWS-Dev environment by changing the following value in `workbc-main/.env`:
PROJECT_ENVIRONMENT=aws-dev
confirm that the active configuration split is now  **AWS Dev**

enter the following commands
`drush cr`
`drush cim`
`drush cr`

Devel module should be enabled
Configuration Ignore setting should now contain  `backup_migrate.*`


change to AWS-Test environment by changing the following value in `workbc-main/.env`:
PROJECT_ENVIRONMENT=aws-test
confirm that the active configuration split is now  **AWS Test**

enter the following commands
`drush cr`
`drush cim`
`drush cr`

Devel module should disabled
Configuration Ignore setting should contain  `backup_migrate.*`


change to AWS-Prod environment by changing the following value in `workbc-main/.env`:
PROJECT_ENVIRONMENT=aws-prod
confirm that the active configuration split is now  **AWS Prod**

enter the following commands
`drush cr`
`drush cim`
`drush cr`

Devel module should disabled
Configuration Ignore setting should contain  `backup_migrate.*`


change to Dev environment by changing the following value in `workbc-main/.env`:
PROJECT_ENVIRONMENT=dev
confirm that the active configuration split is now  **Dev**

enter the following commands
`drush cr`
`drush cim`
`drush cr`

Devel module should enabled
Configuration Ignore setting should be empty
